test/builtin_dataset_mocks.py

```diff
@@ -0,0 +1,492 @@
+                    alternative_hint=lambda _: "",
+                )
+            )
+        self._mock_data_fns[name] = mock_data_fn
+        return mock_data_fn
+
+    def _parse_mock_info(self, mock_info, *, name):
+        if mock_info is None:
+            raise pytest.UsageError(
+                f"The mock data function for dataset '{name}' returned nothing. It needs to at least return an integer "
+                f"indicating the number of samples for the current `config`."
+            )
+        elif isinstance(mock_info, int):
+            mock_info = dict(num_samples=mock_info)
+        elif not isinstance(mock_info, dict):
+            raise pytest.UsageError(
+                f"The mock data function for dataset '{name}' returned a {type(mock_info)}. The returned object should "
+                f"be a dictionary containing at least the number of samples for the current `config` for the key "
+                f"`'num_samples'`. If no additional information is required for specific tests, the number of samples "
+                f"can also be returned as an integer."
+            )
+        elif "num_samples" not in mock_info:
+            raise pytest.UsageError(
+                f"The dictionary returned by the mock data function for dataset '{name}' must contain a `'num_samples'` "
+                f"entry indicating the number of samples for the current `config`."
+            )
+        return mock_info
+
+    def _get(self, dataset, config):
+        name = dataset.info.name
+        resources_and_mock_info = self._cache.get((name, config))
+        if resources_and_mock_info:
+            return resources_and_mock_info
+
+        try:
+            fakedata_fn = self._mock_data_fns[name]
+        except KeyError:
+            raise pytest.UsageError(
+                f"No mock data available for dataset '{name}'. "
+                f"Did you add a new dataset, but forget to provide mock data for it? "
+                f"Did you register the mock data function with `@DatasetMocks.register_mock_data_fn`?"
+            )
+
+        root = self._tmp_home / name
+        root.mkdir(exist_ok=True)
+        mock_info = self._parse_mock_info(fakedata_fn(dataset.info, root, config), name=name)
+
+        mock_resources = []
+        for resource in dataset.resources(config):
+            path = root / resource.file_name
+            if not path.exists() and path.is_file():
+                raise pytest.UsageError(
+                    f"Dataset '{name}' requires the file {path.name} for {config}, but this file does not exist."
+                )
+
+            mock_resources.append(datasets.utils.LocalResource(path))
+
+        self._cache[(name, config)] = mock_resources, mock_info
+        return mock_resources, mock_info
+
+    def _decoder(self, dataset_type):
+        if dataset_type == datasets.utils.DatasetType.RAW:
+            return datasets.decoder.raw
+        else:
+            return lambda file: file.close()
```

Comment:
@bjuncek This is the decoder we are passing to each dataset during the tests unless we manually specify a different one. Thus, although the video datasets do not use a decoder by default through the normal API, they will use one during test.